### ISSUE TITLE
Add support for defaulProps

### ISSUE THREAD

**Issue by NejcZdovc** [palantir/tslint#4130]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.0.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
import * as React from "react"

interface Props {
  variant?: 'one' | 'two'
}

export default class Test extends React.PureComponent<Props, {}> {
  static defaultProps = {
    variant: 'one'
  }

  get something() {
    return {
      one: 'John',
      two: 'Mr. Smith'
    }[this.props.variant]
  }

  render() {
    return <div>Hello ${this.something}!</div>
  }
}
```

with `tslint.json` configuration:

```json
{
  "extends": ["tslint-react", "tslint-config-standard"],
  "linterOptions": {
    "exclude": [
      "package/"
    ]
  },
  "rules": {
    "jsx-no-multiline-js": false
  }
}
```

#### Actual behavior
`TS2538: Type 'undefined' cannot be used as an index type.`
#### Expected behavior
Should work, because we defined variant prop in defaultProp

This feature was introduced with TS 3.0 https://blogs.msdn.microsoft.com/typescript/2018/07/30/announcing-typescript-3-0/#default-props-support

**Comment by giladgray** [comment]
@NejcZdovc this is a `tsc` error, not a tslint warning. all typescript error codes follow the pattern `TS2538`, try googling that error code.
